examples/speech_recognition/modules/time_stretch.py

```diff
@@ -0,0 +1,56 @@
+import torch
+import torch.nn as nn
+import numpy as np
+import random
+import copy
+
+class TimeStretch(nn.Module):
+
+    def __init__(self, w, low, high):
+        super().__init__()
+        if w < 1:
+            raise ValueError("w must be greater than 1")
+        self.w = w
+        self.low = low
+        self.high = high
+
+    def forward(self, batch):
+        new_batch = copy.deepcopy(batch)
+        tokens, lengths = [], []
+        for elem, length in zip(batch['net_input']['src_tokens'], batch['net_input']['src_lengths']):
+            tokens += [time_stretch_seq(elem[:length, :], self.w, self.low, self.high)]
+            lengths += [tokens[-1].size(0)]
+
+        frames = torch.zeros([len(lengths), max(lengths), tokens[0].size(1)], dtype=torch.float32)
```

Comment:
shall we take the `dtype` of the input `src_tokens`? SO this would work also with 16bit fp.